lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4127,15 +4127,38 @@ LLVMToSPIRVBase::transBuiltinToInstWithoutDecoration(Op OC, CallInst *CI,
     // Literal S Literal I Literal rI Literal Q Literal O
 
     Type *ResTy = CI->getType();
-    SPIRVValue *Input =
-        transValue(CI->getOperand(0) /* A - integer input of any width */, BB);
-
-    std::vector<Value *> Operands = {
-        CI->getOperand(1) /* S - bool value, indicator of signedness */,
-        CI->getOperand(2) /* I - location of the fixed-point of the input */,
-        CI->getOperand(3) /* rI - location of the fixed-point of the result*/,
-        CI->getOperand(4) /* Quantization mode */,
-        CI->getOperand(5) /* Overflow mode */};
+    SPIRVValue *Input;
+    std::vector<Value *> Operands(5);
+
+    // If return type of intrinsic is greater than 64 then FE moves return
+    // variable into arguments list
+    if (CI->hasStructRetAttr()) {
```

Comment:
To avoid code duplication I suggest to introduce extra variable:
`uint32_t FirstOperand = 0;`

If we have 'sret' attribute, under
`if (CI->hasStructRetAttr()) {`
we can set it to be
`FirstOperand = 1;`
later it can be used like this:
```
      Operands[0] =
          CI->getOperand(1 + FirstOperand); /* S - bool value, indicator of signedness */
```
so we will have a unified code for 'sret' and 'no sret' cases.

Reply:
but yet some lambda magic from `ArbitraryFloat` handling can (should?) be embraced here.
```
    std::transform(OpItr, OpEnd, std::back_inserter(Literals), [](auto *O) {
      return cast<llvm::ConstantInt>(O)->getZExtValue();
    });
```